src/reduction/mod.rs

```diff
@@ -18,10 +18,10 @@ pub(crate) use crate::bit_depth::reduce_bit_depth;
 /// Attempt to reduce the number of colors in the palette
 /// Returns `None` if palette hasn't changed
 pub fn reduced_palette(png: &PngImage, optimize_alpha: bool) -> Option<PngImage> {
-    if png.ihdr.color_type != ColorType::Indexed {
+    let ColorType::Indexed { palette } = &png.ihdr.color_type else {
         // Can't reduce if there is no palette
         return None;
-    }
+    };
```

Comment:
This is the first time I see refutable patterns in `let` statements being used in practice, neat! :joy: